pcsx2/DEV9/InternalServers/DNS_Logger.cpp

```diff
@@ -0,0 +1,203 @@
+/*  PCSX2 - PS2 Emulator for PCs
+ *  Copyright (C) 2002-2021  PCSX2 Dev Team
+ *
+ *  PCSX2 is free software: you can redistribute it and/or modify it under the terms
+ *  of the GNU Lesser General Public License as published by the Free Software Found-
+ *  ation, either version 3 of the License, or (at your option) any later version.
+ *
+ *  PCSX2 is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ *  PURPOSE.  See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with PCSX2.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "PrecompiledHeader.h"
+
+#include "DNS_Logger.h"
+#include "../PacketReader/IP/UDP/UDP_Packet.h"
+#include "../PacketReader/IP/UDP/DNS/DNS_Packet.h"
+
+using PacketReader::PayloadPtr;
+using namespace PacketReader::IP;
+using namespace PacketReader::IP::UDP;
+using namespace PacketReader::IP::UDP::DNS;
+
+namespace InternalServers
+{
+	void DNS_Logger::InspectRecv(IP_Payload* payload)
+	{
+		UDP_Packet* udpPacket = static_cast<UDP_Packet*>(payload);
+		PayloadPtr* udpPayload = static_cast<PayloadPtr*>(udpPacket->GetPayload());
+		DNS_Packet dns(udpPayload->data, udpPayload->GetLength());
+		LogPacket(&dns);
+	}
+
+	void DNS_Logger::InspectSend(IP_Payload* payload)
+	{
+		UDP_Packet* udpPacket = static_cast<UDP_Packet*>(payload);
+		PayloadPtr* udpPayload = static_cast<PayloadPtr*>(udpPacket->GetPayload());
+		DNS_Packet dns(udpPayload->data, udpPayload->GetLength());
+		LogPacket(&dns);
+	}
+
+	void DNS_Logger::LogPacket(DNS_Packet* dns)
+	{
+		Console.WriteLn("DEV9: DNS: ID %i", dns->id);
+		Console.WriteLn("DEV9: DNS: Is Response? %s", dns->GetQR() ? "True" : "False");
+		const char* opName;
+		switch ((DNS_OPCode)dns->GetOpCode())
```

Comment:
To reduce function size of LogPacket this switch case (and the next) could be broken out to static functions returning `const char*`. (e.g. `DNS_OpCodeToString(opcode)`)